### ISSUE TITLE
ES6 syntax

### ISSUE THREAD

**Issue by petermakowski** [d4nyll/lethargy#17]
Hey @d4nyll, I am going to use lethargy in my project (thanks again, you're a life saver) and have started converting it to ES6 syntax  (it was the only option due to my current setup). I was wondering what are your plans regarding the library, as I know you plan to refactor https://github.com/d4nyll/smartscroll to ES6 at some point. 

I will gladly share my work here (translate to ES6 and add Babel to transpile to ES5) if you're willing to move away from coffee script. Please let me know your thoughts.

**Comment by d4nyll** [comment]
Hi @petermakowski I would definitely appreciate it if you can port it to ES6. The only reason I used CoffeeScript was because it supported classes when ES6 was not so popular.

I'll add you as a collaborator on this project, which means you can create a new branch on this repository and push your work on it, without having to fork it and do pull request.

**Comment by d4nyll** [comment]
Hey @petermakowski I see you've ported the code using decaffeinate. Do you mind if I carry on with it and refactor it some more? I got a few people who are available at work to refactor it as an exercise.

**Comment by petermakowski** [comment]
Absolutely, feel free to do so - I'm not 100% happy with how it looks yet, your original coffeescript was much cleaner and minified JavaScript was smaller. I not only converted it using decaffeinate but fixed multiple issues in the code manually as well. I planned to improve it more, but I'm busy with other projects at the moment. I was thinking of using Array map for coffeescript ranges, but that would need testing as apparently it's slower (havent' tried it myself) - https://github.com/hemanth/coffeescript-equivalents-in-es6#ranges

**Comment by d4nyll** [comment]
We've made a few changes, I'll clean it up a little and push it on Monday.

**Comment by petermakowski** [comment]
Hey, I made quite a few changes, please review (look at the commit message). Could you let me know what else would you need before merging to the master branch?

There is definitely still some bugfixing to do as I noticed that since this commit https://github.com/d4nyll/lethargy/commit/58382fc161241ce21e413155508b56ee5875e2bd Firefox keeps returning false after first 2 scrolls no matter what you do (tested with Macbook's trackpad).

I can look into it if you'd like.

**Comment by d4nyll** [comment]
There were some changes I wanted to make to the code, such as making the defaults available as a static property of the class. I'll make those changes over the weekend.

After that, you can take a final look and we can then test it on different browsers at the beginning of next week, fix any bugs and merge it into master next weekend.

How's that sound?

**Comment by petermakowski** [comment]
Sounds good! Actually I managed to fix Firefox issue just now https://github.com/d4nyll/lethargy/commit/332fe29b73dcce58cae9234aee1d8046ca9b66fb.
